src/createSelectorCreator.ts

```diff
@@ -408,23 +450,28 @@ export function createSelectorCreator<
         arguments
       )
 
-      if (
-        process.env.NODE_ENV !== 'production' &&
-        shouldRunInputStabilityCheck(inputStabilityCheck, firstRun)
-      ) {
-        // make a second copy of the params, to check if we got the same results
-        const inputSelectorResultsCopy = collectInputSelectorResults(
-          dependencies,
-          arguments
-        )
+      if (process.env.NODE_ENV !== 'production') {
+        if (shouldRunDevModeCheck(identityFunctionCheck, firstRun)) {
+          runIdentityFunctionCheck(
+            resultFunc as Combiner<InputSelectors, Result>
+          )
+        }
+
+        if (shouldRunDevModeCheck(inputStabilityCheck, firstRun)) {
+          // make a second copy of the params, to check if we got the same results
+          const inputSelectorResultsCopy = collectInputSelectorResults(
+            dependencies,
+            arguments
+          )
 
-        runStabilityCheck(
-          { inputSelectorResults, inputSelectorResultsCopy },
-          { memoize, memoizeOptions: finalMemoizeOptions },
-          arguments
-        )
+          runStabilityCheck(
+            { inputSelectorResults, inputSelectorResultsCopy },
+            { memoize, memoizeOptions: finalMemoizeOptions },
+            arguments
+          )
 
-        if (firstRun) firstRun = false
+          if (firstRun) firstRun = false
+        }
```

Comment:
```suggestion
        }
        if (firstRun) firstRun = false
```

Reply:
currently you're only reassigning the firstRun variable inside the if block for the stability check - this means that if you have stability check disabled firstRun will always be true

Reply:
Oh wow, can't believe I that, thanks for letting me know.